content/blog/grpc-web-interceptor.md

```diff
@@ -1,46 +1,30 @@
 ---
-title: Using gRPC-Web Interceptors
-date: 2020-06-01
+title: Interceptors in gRPC-Web
+date: 2020-06-11
 authors:
 - name: Zhenli Jiang
   position: Google
 ---
 
-Interceptor is a very common pattern for RPC client APIs in many languages.
-gRPC-Web has also released this feature recently (May 2020 v1.1.0). The current
-design is based on other gRPCs’ (Java, Node, Go, etc) client interceptors, with
-some gRPC-Web specific features for better developer experience. Meanwhile,
-we’ve made a lot of effort to integrate gRPC-Web and the client interceptor
-design into Google’s internal Web frameworks. We believe that our current design
-allows gRPC-Web to be easily integrated into modern Web frameworks and the API
-is also very easy for gRPC-Web users to adopt.
+We're pleased to announce support for _interceptors_ in [gRPC-web][] as of
+release [1.1.0][]. While the current design is based on gRPC client interceptors
+available from other [gRPC languages][], it also includes gRPC-web specific
+features that should make interceptors easy to adopt and use alongside modern
+web frameworks.
 
 ## Introduction
 
-In [gRPC-Web](https://github.com/grpc/grpc-web) there are two kinds of [client
-interceptors](https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/interceptor.js).
-The type of interceptors is determined by the response type of the RPC call.
+Similar to other gRPC languages, gRPC-web supports _unary_ and server
+_streaming_ interceptors. For each kind of interceptor, we've [defined an
+interface][interceptor.js] containing a single `intercept()` method:
 
-- **UnaryInterceptor**: for Promise responses
-- **StreamInterceptor**: for ClientReadableStream responses
+- `UnaryInterceptor`
+- `StreamInterceptor`
 
-gRPC-Web has two clients generated by
-[Protoc](https://github.com/grpc/grpc-web#client-configuration-options) for one
-service: let’s say *FooServiceClient* and *FooServicePromiseClient*. The
-response type of *FooServiceClient* is
-[ClientReadableStream](https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/clientreadablestream.js)
-for BOTH unary calls and server streaming calls, so **StreamInterceptor** is
-expected to be used for intercepting *FooServiceClient* calls. The response type
-of *PromiseClient* is
-[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise),
-so we should use **UnaryInterceptor** for *PromiseClients*.
-
-The design of UnaryInterceptor looks like:
+For example, the `UnaryInterceptor` interface is declared like this:
 
 ```js
 /*
-* StreamInterceptor shares the same semantics except the return type of it is
-* ClientReadablaStream
```

Comment:
I moved this comment into the main text (leaving it hear seems too early and it feels more confusing than anything else).